tests/test_discover_api.py

```diff
@@ -1604,6 +1604,53 @@ def test_exception_stack_column_boolean_condition(self) -> None:
         data = json.loads(response.data)
         assert data["data"] == [{"count": 1}]
 
+    def test_extra_group_conditions(self) -> None:
+        response = self.post(
+            json.dumps(
+                {
+                    "selected_columns": [
+                        "title",
+                        "type",
+                        [
+                            "coalesce",
+                            ["email", "username", "ip_address"],
+                            "user.display",
+                        ],
+                        "timestamp",
+                        "project_id",
+                        "event_id",
+                        [
+                            "transform",
+                            [
+                                ["toString", ["project_id"]],
+                                ["array", [f"'{self.project_id}'"]],
+                                ["array", ["'my-project'"]],
+                                "''",
+                            ],
+                            "project",
+                        ],
+                    ],
+                    "having": [],
+                    "orderby": ["-timestamp"],
+                    "limit": 51,
+                    "offset": 0,
+                    "project": [self.project_id],
+                    "dataset": "discover",
+                    "from_date": (self.base_time - self.skew).isoformat(),
+                    "to_date": (self.base_time + self.skew).isoformat(),
+                    "groupby": [],
+                    "conditions": [["project_id", "IN", [self.project_id]]],
+                    "aggregations": [],
+                    "consistent": False,
+                    "debug": False,
+                }
+            ),
+            entity="discover",
+        )
+        assert response.status_code == 200
+        data = json.loads(response.data)
+        assert "PREWHERE in(event_id, tuple" in data["sql"]
```

Comment:
Can we assert on the entire PREWHERE expression?

Reply:
If you'd rather, I can do that.

Reply:
I don't think this is a great approach to do this test.
First: I am not sure, by reading the test name `test_extra_group_conditions`, what this is supposed to test. We could also make a case that splitter is an implementation detail that should not be tested by an integration test that only checks that the output matches what is expected for an input.
Second we are depending on the SQL parameter in the response which we know is broken. It does not reflect the splitter (it includes only one of the queries).

If we are looking for a specific issue with the splitter, can't we build a splitter unit test ?